### PR TITLE
Enable branch-wise override of distro map values

### PR DESCRIPTION
**Pull request by nhr**

### CODE REVIEW

**Comment by nhr** [comment]
@Fryguy please review - this is high priority for the OpenShift docs team.

File: lib/ascii_binder/helpers.rb [hunk 1]
```diff
@@ -748,6 +748,24 @@ def configure_and_generate_page options
       single_page    = options[:single_page]
       site_name      = options[:site_name]
 
+      # Distro Map settings can be overridden on a per-branch
+      # basis. This only works for top-level (string) values
+      # of the distro config and -not- the 'site' key.
+      branchwise_distro_config = {}
+      distro_config.each do |key,value|
+        next unless distro_config[key].class == String
```
Comment (Fryguy) [inline lib/ascii_binder/helpers.rb]: Prefer `next unless distro_config[key].kind_of?(String)`

**Comment by Fryguy** [comment]
Can you just post an example here of what it would look like in the distro_map?  Eventually, we'll probably want to document this feature in the asciibinder documention

**Comment by nhr** [comment]
Here's an example; this is the actual plan to update the OpenShift docs distro map:

```
openshift-enterprise:
  name: OpenShift Container Platform
  author: OpenShift Documentation Project <dev@lists.openshift.redhat.com>
  site: commercial
  site_name: Documentation
  site_url: https://docs.openshift.com/
  branches:
    enterprise-3.0:
      name: '3.0'
      dir: enterprise/3.0
      distro-overrides:
        name: OpenShift Enterprise
    enterprise-3.1:
      name: '3.1'
      dir: enterprise/3.1
      distro-overrides:
        name: OpenShift Enterprise
    enterprise-3.2:
      name: '3.2'
      dir: enterprise/3.2
      distro-overrides:
        name: OpenShift Enterprise
    master:
      name: '3.3'
      dir: container-platform/3.3

```

**Comment by vikram-redhat** [comment]
@nhr @Fryguy  - what's the likely time-frame for this now?

**Comment by nhr** [comment]
@vikram-redhat - I want to add one other enhancement but my plan is to roll all of this out today.

**Comment by nhr** [comment]
Actually - I don't want to bloat this PR; I get things rolling now. I reached out to Jason yesterday to let him know I needed to merge this today, so I will proceed.

**Pull request merged**